lms/djangoapps/learner_recommendations/tests/test_utils.py

```diff
@@ -103,11 +146,49 @@ def test_enrolled_courses_are_removed_from_recommendations(
         """
         total_enrolled_courses = 6
         total_recommendations = len(self.recommended_course_keys)
-        mocked_get_course_data.return_value = self.course_data
+        mocked_get_course_data.side_effect = self._mock_get_course_data
         for course_run_key in self.course_run_keys[:total_enrolled_courses]:
             CourseEnrollmentFactory(course_id=course_run_key, user=self.user)
 
         filtered_courses = filter_recommended_courses(
             self.user, self.recommended_course_keys, total_recommendations
         )
-        self.assertEqual(len(filtered_courses), total_recommendations - total_enrolled_courses)
+        assert len(filtered_courses) == (total_recommendations - total_enrolled_courses)
+
+    @patch("lms.djangoapps.learner_recommendations.utils.get_course_data")
```

Comment:
It is recommended to patch/mock from where the module is used rather than where it is defined.

Reply:
Yeah I am mocking the place where it is used.
https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/learner_recommendations/utils.py#L240

It is defined in catalog utils which I am not mocking.

Reply:
Thanks for the clerification